### PR TITLE
Test: add support for RHEL guest

### PR DESCRIPTION
**Pull request by dperpeet**

### CODE REVIEW

File: tools/make-rpms [hunk 1]
```diff
@@ -77,16 +77,39 @@ rm -rf $base/mock/*
...
 """
 EOF
-# HACK: Signatures seem to be unreliable in Fedora 21 right now
-sed -i -e 's/gpgcheck=1/gpgcheck=0/' $base/mock/$os-$arch-cockpit.cfg
+fi
+
+if [ "$os"="fedora-21" ]; then
+    # HACK: Signatures seem to be unreliable in Fedora 21 right now
+    sed -i -e 's/gpgcheck=1/gpgcheck=0/' $base/mock/$os-$arch-cockpit.cfg
+fi
```
Comment (stefwalter) [inline tools/make-rpms]: This is gonna break pretty regularly given that Fedora doesn't sign stuff until pretty late in the release cycle.

File: test/testvm.py [hunk 2]
```diff
@@ -83,10 +83,10 @@ def stop(self):
 
     def wait_ssh(self):
         """Try to connect to self.address on port 22"""
-        tries_left = 5
+        tries_left = 60
         while (tries_left > 0):
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            sock.settimeout(2)
+            sock.settimeout(1)
```
Comment (stefwalter) [inline test/testvm.py]: Separate commit?

File: test/testvm.py [hunk 3]
```diff
@@ -423,6 +423,26 @@ def _setup_fedora_22 (self, gf):
...
+        # systemctl disable sshd.service
+        gf.rm_f("/etc/systemd/system/multi-user.target.wants/sshd.service")
+        # systemctl enable sshd.socket
+        gf.mkdir_p("/etc/systemd/system/sockets.target.wants/")
+        gf.ln_sf("/usr/lib/systemd/system/sshd.socket", "/etc/systemd/system/sockets.target.wants/")
+
+        # upload subscription information
+        gf.mkdir_p("/root/.rhel")
+        gf.upload(os.path.expanduser("~/.rhel/login"), "/root/.rhel/login")
+        gf.upload(os.path.expanduser("~/.rhel/pass"), "/root/.rhel/pass")
```
Comment (stefwalter) [inline test/testvm.py]: Will we end up embedding our personal login credentials in our images?

Comment (dperpeet) [inline test/testvm.py]: Waiting for response to my request for cockpit-team access credentials.

File: test/cockpit-rhel-7.setup [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+COCKPIT_DEPS="accountsservice-libs udisks2 libudisks2 json-glib realmd lvm2 mdadm glib-networking libssh storaged selinux-policy-targeted docker pcp-libs pcp kubernetes subscription-manager expect"
+# We also install the packages necessary to join a FreeIPA domain so
+# that we don't have to go to the network during a test run.
+IPA_CLIENT_PACKAGES="freeipa-client oddjob oddjob-mkhomedir sssd"
+
+TEST_PACKAGES="systemtap-runtime-virtguest"
+
+yum install --nogpgcheck -y $TEST_PACKAGES $COCKPIT_DEPS $IPA_CLIENT_PACKAGES
+
+subscription-manager unregister
```
Comment (mvollmer) [inline test/cockpit-rhel-7.setup]: We should remove `~/.rhel/*` after this, no?

File: test/cockpit-rhel-7.setup [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+# Stopping a user@.service at poweroff sometimes hangs and then times
+# out, but that seems to be harmless otherwise.  We reduce the timeout
+# so that we don't have to wait for the default 90 seconds.
+#
+f=/usr/lib/systemd/system/user@.service
+if [ -f $f ] && ! grep -q TimeoutStopSec $f; then
+  echo TimeoutStopSec=1 >>$f
+  systemctl daemon-reload
+fi
```
Comment (mvollmer) [inline test/cockpit-rhel-7.setup]: Let's remove this hack until we know that we need it.

File: test/cockpit-rhel-7.setup [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+yum -y install wget yum-utils
+cd /etc/yum.repos.d
+wget -T 15 -t 4 https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/epel-7/sgallagh-cockpit-preview-epel-7.repo
+yum-config-manager --enable rhel-7-server-optional-rpms
+yum-config-manager --enable rhel-7-server-extras-rpms
+yum-config-manager --add-repo=http://dl.fedoraproject.org/pub/epel/7/x86_64/
+yum -y update
+cd /tmp/dep
+yum-builddep --nogpgcheck -y cockpit.spec
```
Comment (mvollmer) [inline test/cockpit-rhel-7.setup]: Why do we install builddeps into the image?

File: test/cockpit-rhel-7.setup [hunk 7]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -e
+
+# register system
+subscription-manager register --auto-attach --username=`cat ~/.rhel/login` --password=`cat ~/.rhel/pass`
+
+yum -y install wget yum-utils
+cd /etc/yum.repos.d
+wget -T 15 -t 4 https://copr.fedoraproject.org/coprs/sgallagh/cockpit-preview/repo/epel-7/sgallagh-cockpit-preview-epel-7.repo
```
Comment (mvollmer) [inline test/cockpit-rhel-7.setup]: What do we need this for?  Newer glib?

File: test/testvm.py (hunk 2 above)
Comment (dperpeet) [inline test/testvm.py]: this is now already in master: https://github.com/cockpit-project/cockpit/commit/878cd8547335171fffd7de2b8b9a3b3a52bdf606

File: test/cockpit-rhel-7.setup (hunk 4 above)
Comment (dperpeet) [inline test/cockpit-rhel-7.setup]: yes, good catch - done via `rm -rf ~/.rhel`

File: test/cockpit-rhel-7.setup (hunk 5 above)
Comment (dperpeet) [inline test/cockpit-rhel-7.setup]: done, along with a bit of other cleanup

File: test/cockpit-rhel-7.setup (hunk 6 above)
Comment (dperpeet) [inline test/cockpit-rhel-7.setup]: we don't need them!

**Comment by dperpeet** [comment]
needs to be tested against rhel image by hubbot

**Comment by mvollmer** [comment]
> needs to be tested against rhel image by hubbot

I have installed the rhel images for hubbot and the testsuite runs.  Hubbot will soon learn to test master with TEST_OS=rhel-7.

**Pull request closed without merging**